frontend/packages/kubevirt-plugin/src/components/create-vm/forms/boot-source-form.tsx

```diff
@@ -0,0 +1,305 @@
+import ProjectDropdown from '../../form/ProjectDropdown';
+import { EXAMPLE_CONTAINER, FEDORA_IMAGE_LINK, RHEL_IMAGE_LINK } from '../../../utils/strings';
+import { getDefaultStorageClass } from '../../../selectors/config-map/sc-defaults';
+import { BootSourceAction, BootSourceState, BOOT_ACTION_TYPE } from './boot-source-form-reducer';
+import { DataSource } from './data-source';
+import { AccessMode } from '../../../constants';
+
+type BootSourceFormProps = {
+  state: BootSourceState;
+  dispatch: React.Dispatch<BootSourceAction>;
+  withUpload?: boolean;
+};
+
+export const BootSourceForm: React.FC<BootSourceFormProps> = ({ state, dispatch, withUpload }) => {
+  const [isDataSourceOpen, setDataSourceOpen] = React.useState(false);
+  const [showAdvanced, setShowAdvanced] = React.useState(false);
+
+  const [storageClasses, scLoaded] = useK8sWatchResource<StorageClassResourceKind[]>({
+    kind: StorageClassModel.kind,
+    isList: true,
+    namespaced: false,
+  });
+
+  const useProjects = useFlag(FLAGS.OPENSHIFT);
+
+  const defaultSCName = getDefaultStorageClass(storageClasses)?.metadata.name;
+
+  const handleStorageClass = (scName: string) => {
+    const updatedStorageClass = storageClasses.find((sc) => sc.metadata.name === scName);
+    const provisioner = updatedStorageClass?.provisioner || '';
+    const modes =
+      provisionerAccessModeMapping[provisioner] || getAccessModeForProvisioner(provisioner);
+    dispatch({ type: BOOT_ACTION_TYPE.SET_ACCESS_MODE, payload: AccessMode.READ_WRITE_ONCE });
+    dispatch({
+      type: BOOT_ACTION_TYPE.SET_ACCESS_MODES,
+      payload: modes.map((m) => AccessMode.fromString(m)),
+    });
+    dispatch({
+      type: BOOT_ACTION_TYPE.SET_STORAGE_CLASS,
+      payload: updatedStorageClass?.metadata?.name,
+    });
+  };
+
+  if (scLoaded && !state.storageClass?.value) {
+    handleStorageClass(defaultSCName);
+  }
+
+  const isUpstream = window.SERVER_FLAGS.branding === 'okd';
+  return (
+    <Form>
+      <FormRow fieldId="form-data-source" title="Data source type" isRequired>
+        <Select
+          placeholderText="--- Select data source ---"
+          aria-label="Select data source"
+          onToggle={setDataSourceOpen}
+          onSelect={(e, value) => {
+            dispatch({
+              type: BOOT_ACTION_TYPE.SET_DATA_SOURCE,
+              payload: value as DataSource,
+            });
+            setDataSourceOpen(false);
+          }}
+          selections={state.dataSource?.value}
+          isOpen={isDataSourceOpen}
+        >
+          {DataSource.getAll()
+            .filter((ds) => (withUpload ? true : ds !== DataSource.UPLOAD))
+            .map((ds) => (
+              <SelectOption key={ds.getValue()} value={ds} description={ds.getDescription()} />
+            ))}
+        </Select>
+      </FormRow>
+      {state.dataSource?.value === DataSource.UPLOAD && (
+        <FormRow fieldId="form-ds-file" title="Upload source" isRequired>
+          <FileUpload
```

Comment:
we should add type restrictions and onDropReject handlers

```js
        dropzoneProps={{
            accept: '.iso,.img,.qcow2,.gz,.xz',
            onDropRejected: () => setIsFileRejected(true),
            onDropAccepted: () => setIsFileRejected(false),
          }}
```

Reply:
done

Reply:
About the filetypes, heads up:
https://bugzilla.redhat.com/show_bug.cgi?id=1894374